catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -215,8 +222,88 @@ private org.apache.hadoop.hive.metastore.api.Partition toHivePartition(
   }
 
   @Override
-  public boolean dropPartition(String partitionName) {
-    throw new UnsupportedOperationException();
+  public boolean dropPartition(String partitionName, boolean ifExists)
+      throws NoSuchPartitionException {
+    try {
+      // check the partition exists
+      try {
+        table
+            .clientPool()
+            .run(c -> c.getPartition(table.schemaName(), table.name(), partitionName));
+      } catch (NoSuchObjectException e) {
+        if (ifExists) {
+          return true;
+        } else {
+          throw new NoSuchPartitionException(
+              e, "Hive partition %s does not exist in Hive Metastore", partitionName);
+        }
+      }
+
+      // get all partitions that will be deleted
```

Comment:
Is it necessary to get partitions from HMS? Why not just call the HMS dropPartition method using partitionName?

Reply:
HMS dropPartition method only support drop one partition with a complete name
for example, there are partitions:
log_date=0312/log_hour=10
log_date=0312/log_hour=11
log_date=0312/log_hour=12

If you deliver "log_date=0312" to HMS dropPartition method, it will report error.
So you have to get all partitions from HMS whose parent partition is "log_date=0312", and drop them one by one.
You can see what  I mentioned in @yuqi1129 's discussion, both Spark and Hcat implement dropPartition in this way

Reply:
If that's the case, please add comments to the implementations

Reply:
OK, nice advice